src/server/api/controllers/profile.controller.js

```diff
@@ -37,7 +37,31 @@ const getAdminsProfile = async (userId) => {
     return error.message;
   }
 };
+
+const postAdminsProfile = async (adminRole, userId) => {
+  try {
+    const userProfileData = await knex.raw(
+      ` SELECT us.id AS userID, us.name as userName FROM users AS us, user_roles AS ur, roles AS rl WHERE ur.user_id = us.id AND ur.role_id = rl.id AND rl.name = 'student' AND us.id = ? `,
+      [userId],
+    );
+    if (userProfileData[0].length === 0) {
+      throw new Error(`incorrect entry with the id of ${userId}`, 404);
+    } else {
+      await knex('users').where({ id: userId }).update({
+        name: adminRole.name,
+      });
+
+      await knex('user_roles').where({ id: userId }).update({
+        role_id: adminRole.role,
+      });
```

Comment:
You are supposed to update the property `admin_role` in the `users` table, not the role in `user_roles`.